storage/shard.go

```diff
@@ -1529,6 +1617,71 @@ func (s *dbShard) removeAnyFlushStatesTooEarly() {
 	s.flushState.Unlock()
 }
 
+func (s *dbShard) SnapshotState() (bool, time.Time) {
+	s.snapshotState.RLock()
+	defer s.snapshotState.RUnlock()
+	return s.snapshotState.isSnapshotting, s.snapshotState.lastSuccessfulSnapshot
+}
+
+func (s *dbShard) markIsSnapshotting() {
+	s.snapshotState.Lock()
+	s.snapshotState.isSnapshotting = true
+	s.snapshotState.Unlock()
+}
+
+func (s *dbShard) markDoneSnapshotting(success bool, completionTime time.Time) {
+	s.snapshotState.Lock()
+	s.snapshotState.isSnapshotting = false
+	if success {
+		s.snapshotState.lastSuccessfulSnapshot = completionTime
+	}
+	s.snapshotState.Unlock()
+}
+
+// TODO(rartoul): Move into cleanup manager
```

Comment:
^^def do this before merging

Reply:
Prolly gonna be a separate P.R cause this one is already too big, but I made an issue for it

Reply:
https://github.com/m3db/m3db/issues/509